src/components/Modal.js

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from "react";
+import { TaskType as TaskType } from "../TaskType";
+import { v4 as uuidv4 } from "uuid";
+
+const Modal = ({ handleClose, addTask, isEditing, currTask, updateTask, taskType}) => {
+  const [taskName, setTaskName] = React.useState("");
+  const [state, setState] = React.useState(taskType);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!taskName) return;
+    if(!isEditing){
+      const newTask = { name: taskName, id: uuidv4(), date: Date.now(), state };
+      addTask(newTask);
+    }
+    else {
+      const updatedTask = {
+        ...currTask,
+        name : taskName,
+        state : state,
+      }
+      updateTask(updatedTask);
+    }
+    handleClose();
+  };
+
+  const handleState = (e) => {
+    setState(e.target.value);
+  };
+
+  useEffect(() => {
```

Comment:
`useEffect` should be at the top of the component. Makes it easier to understand that it is performing some lifecycle method change.